### ISSUE TITLE
BBR transmits too aggressively under throttled network

### ISSUE THREAD

**Issue by keytouch** [apernet/hysteria#1180]
**Describe the bug**
With normal ISP broadband with 30Mbps up, speedtest utilize about 70Mbps with BBR. While effective bandwidth is of course within 30Mbps.
Aggressive transmiting behaviour might lead to latency increase and jitters... Maybe there is something odd with congestion window detection.

**To Reproduce**
Just run `hysteria-linux-amd64-avx speedtest`
To mitigate the issue, I have to lower `maxStreamReceiveWindow` or use brutal, which is not even a work around.
I am using 2.5.0.

**Expected behavior**
Transmit rate should not shoot much beyond 30Mbps.

**Comment by its0ka** [comment]
and also kernel TCP BBR on linux works like 10 times better under the same network conditions, i tested on my local network with "tc qdisc replace dev enp2s14 root netem limit 999999999 delay 100ms 50ms loss random 10%" and set network card to 10mbit/s and got
hysteria bbr: 700kbit/s iperf3 (network card was transfering at 1.4mbit/s)
kernel bbr: 8mbit/s iperf3 (network card was transfering at 8.5mbit/s)

but with "tc qdisc replace dev enp2s14 root netem limit 999999999 delay 100ms 00ms loss random 10%" i get much more similar results, seems like hysteria bbr hates jitter

**Comment by BK8000L** [comment]
brutal also seems to have issues, but kernel module with sing-box works much better
`tc qdisc replace dev enp2s14 root netem limit 999999999 delay 100ms 50ms loss random 10%` and brutal at 6mbit/s
sing-box: 712kb/s (network transfer at 6.4mbit/s)
hysteria2: 135kb/s (network transfer at 6.6mbit/s)

**Comment by its0ka** [comment]
nothing new about this? at least #won't fix?

**Comment by its0ka** [comment]
@tobyxdd @haruue please say something at least. There is a lot of wasted bandwidth because of this, internet in my country already collapses in the evening, a lot of people use hy2 and this is not helping it. Is this a hy2 issue or a quic protocol issue?

**Comment by tobyxdd** [comment]
> [@tobyxdd](https://github.com/tobyxdd) [@haruue](https://github.com/haruue) please say something at least. There is a lot of wasted bandwidth because of this, internet in my country already collapses in the evening, a lot of people use hy2 and this is not helping it. Is this a hy2 issue or a quic protocol issue?

Yeah there are some known problems with the current implementation. There have been a few attempts to fix it but none have fully fixed it. At this point I'm just waiting for quic-go's official BBR support.

In the meantime you should use Brutal (by providing bandwidth values) instead if you have problems with BBR.

**Comment by its0ka** [comment]
@tobyxdd brutal is still wasting bandwidth https://github.com/apernet/hysteria/issues/1180#issuecomment-2520612823, in the evening i lose like 40% of bandwidth (network receives at 80mbit/s but curl shows 50mbit/s)
the problem is not packet loss but packet reordering i guess (iperf3 starts to show a lot of datagrams received out of order in the evening)
